### PR TITLE
Update to support stylelint v14.

### PR DESCRIPTION
**Pull request by dbatiste**
This PR updates deps and config with necessary changes to support v14 of stylelint ([Migration Guide](https://github.com/stylelint/stylelint/blob/main/docs/migration-guide/to-14.md#migrating-to-1400)).

It configures syntax handlers for CSS in JS, and also CSS in HTML.  If requested, we can configure to handle other file extensions, and other syntax handlers.

I also removed the `selector-class-pattern` because we no longer support any browsers that do not support native shadowDOM.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -76,7 +76,6 @@ module.exports = {
     "selector-attribute-operator-space-before": "never",
     "selector-combinator-space-after": "always",
     "selector-combinator-space-before": "always",
-    "selector-class-pattern": "(^d2l-)|(^focus-visible$)",
```
Comment (dlockhart) [inline index.js:79]: What does this have to do with shadow DOM? I thought it was defining the CSS class selector pattern that we want folks to use -- i.e. `.d2l-foo` instead of `.foo`.

Comment (dbatiste) [reply index.js:79]: I think we wanted them to do that because of the potential collision that could occur when shadyDOM was in play.  By prefixing their classes with the component name, it was preventing CSS defined in the document from targeting HTML elements in the shadyDOM.  Now that all the components' elements are properly scoped, we should be able to use more succinct, less redundant class names.  If we want to keep the rule for conventional reasons, I don't feel strongly about it, but this linting rule was originally added to avoid collision.

Comment (dbatiste) [reply index.js:79]: By "preventing", what I mean is, it was reducing the likelihood of a collision.

Comment (dlockhart) [reply index.js:79]: Ah I understand. Yeah I guess that's definitely no longer a requirement and I also don't feel super strongly about keeping it around as a convention. 👍

**Review by dlockhart — APPROVED** [review]
